src/raft.rs

```diff
@@ -883,7 +887,7 @@ impl<T: Storage> Raft<T> {
             "term" => self.term
         );
         self.register_vote(self_id, acceptance);
-        if let CandidacyStatus::Elected = self.prs().candidacy_status(&self.votes) {
+        if let CandidacyStatus::Elected = self.prs().candidacy_status(&self.votes, self.quorum_fn) {
```

Comment:
For the election, I think it's better to use `majority` as before because the majority is enough and has no effect on the case described in https://github.com/etcd-io/etcd/issues/11506.

Reply:
Suppose a Raft group in 2 data centers meets 1 data center failure. If the Raft group is recovered from the rest data center manually, there could be 2 leaders if the failed data center also recovers. So it's a disturbance to use 2 quorum algorithm for election and replication I guess. It's better to keep them as same.

Reply:
2 leader case is no problem. As long as the quorum algorithm for election and replication are both greater than the majority, the safety is guaranteed.
(This condition can be weaker according to the flexible paxos paper)

Reply:
This won't introduce inconsistent data, but could introduce unavailable (if clients connect to the partition in failure-recover data center).